### PR TITLE
Add string formatting checks for C# and Java

### PR DESCRIPTION
**Pull request by TylerRupert**
Before submitting pull request, please ensure that:

- [ ] Every commit has message which describes it
- [ ] Every commit is needed on it's own, if you have just minor fixes to previous commits, you can squash them
- [ ] Any code changes come with testcase
- [ ] Any new functionality is covered by user documentation

We have a lot of C# and Java source code and we have found these additional checks to be very valuable in the localization process.

### CODE REVIEW

**Comment by nijel** [comment]
I'm sorry for failing tests in master, it should be fixed now. Can you please rebase on current master?

**Comment by TylerRupert** [comment]
Yeah, no problem. I am out of office until next week, but will take care of this as soon as I return.

File: weblate/checks/tests/test_format_checks.py [hunk 1]
```diff
@@ -409,3 +409,216 @@ def test_wrong_attribute_format(self):
...
+        ))
+        
+    def test_missing_named_format_ignore(self):
+        self.assertFalse(self.check.check_format(
+            '{0} string',
+            'string',
+            True
+        ))
+        
+    def test_escaping(self):
```
[the remark this answers is not in the document]
Comment (nijel) [reply weblate/checks/tests/test_format_checks.py]: Also please address this - the same method is defined twice.

**Comment by TylerRupert** [comment]
I made the changes and squashed then rebased on the current master but it appears I made a mistake somewhere in the process since I am still learning Git and contributing to open source. I will be back in the office on Wednesday to correct this.

**Comment by nijel** [comment]
You seem to have rebased on your master branch instead of upstream one. I've tried to fix this in https://github.com/WeblateOrg/weblate/pull/2175

**Comment by nijel** [comment]
Thanks, I've merged it rebased in https://github.com/WeblateOrg/weblate/pull/2175

**Pull request closed without merging**